From c55383f57e6f8e85c59a23918e870009e9787ddc Mon Sep 17 00:00:00 2001
From: Andrew Kelley <andrew@ziglang.org>
Date: Wed, 3 Sep 2025 15:25:41 -0700
Subject: [PATCH 108/133] std.Progress: avoid problematic catch syntax
X-Developer-Signature: v=1; a=openpgp-sha256; l=2680; i=xtex@astrafall.org;
 h=from:subject; bh=s/Yx5BFk6zjjDDY1qvPVB1zZyRu5MjsBN2CDW+e6WFQ=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78nhi+4N3l3sl+d385Gjc0nW402P7/e9beaVbVP6qb
 Ph889WVjoksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICp3rGd4Z+RhYeg1GMfXtXaN1vippzt3flCKpdVKct88zLxkEq/nhkM/530m4w
 U4jQ07G7q3y4stJGK73KoDbodbhGrUdGjqm7FCQA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 95439c982033bf56a54f4dd85e6de951b52aeb0d)
---
 lib/std/Progress.zig | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/lib/std/Progress.zig b/lib/std/Progress.zig
index 58d409ac07fc..223c7fc58993 100644
--- a/lib/std/Progress.zig
+++ b/lib/std/Progress.zig
@@ -1248,7 +1248,9 @@ fn computeRedraw(serialized_buffer: *Serialized.Buffer) struct { []u8, usize } {
                         i += progress_pulsing.len;
                     } else {
                         const percent = completed_items * 100 / estimated_total;
-                        i += (std.fmt.bufPrint(buf[i..], @"progress_normal {d}", .{percent}) catch &.{}).len;
+                        if (std.fmt.bufPrint(buf[i..], @"progress_normal {d}", .{percent})) |b| {
+                            i += b.len;
+                        } else |_| {}
                     }
                 },
                 .success => {
@@ -1265,7 +1267,9 @@ fn computeRedraw(serialized_buffer: *Serialized.Buffer) struct { []u8, usize } {
                         i += progress_pulsing_error.len;
                     } else {
                         const percent = completed_items * 100 / estimated_total;
-                        i += (std.fmt.bufPrint(buf[i..], @"progress_error {d}", .{percent}) catch &.{}).len;
+                        if (std.fmt.bufPrint(buf[i..], @"progress_error {d}", .{percent})) |b| {
+                            i += b.len;
+                        } else |_| {}
                     }
                 },
             }
@@ -1364,12 +1368,18 @@ fn computeNode(
     if (!is_empty_root) {
         if (name.len != 0 or estimated_total > 0) {
             if (estimated_total > 0) {
-                i += (std.fmt.bufPrint(buf[i..], "[{d}/{d}] ", .{ completed_items, estimated_total }) catch &.{}).len;
+                if (std.fmt.bufPrint(buf[i..], "[{d}/{d}] ", .{ completed_items, estimated_total })) |b| {
+                    i += b.len;
+                } else |_| {}
             } else if (completed_items != 0) {
-                i += (std.fmt.bufPrint(buf[i..], "[{d}] ", .{completed_items}) catch &.{}).len;
+                if (std.fmt.bufPrint(buf[i..], "[{d}] ", .{completed_items})) |b| {
+                    i += b.len;
+                } else |_| {}
             }
             if (name.len != 0) {
-                i += (std.fmt.bufPrint(buf[i..], "{s}", .{name}) catch &.{}).len;
+                if (std.fmt.bufPrint(buf[i..], "{s}", .{name})) |b| {
+                    i += b.len;
+                } else |_| {}
             }
         }
 
-- 
2.52.0

